subsys/caf/modules/shell.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/shell/shell.h>
+#include <app_event_manager.h>
+#include <caf/events/button_event.h>
+
+static int set_button_handler(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc < 3) {
+		shell_error(shell, "Missing required parameter(s)");
+		return -EINVAL;
+	}
+
+	struct button_event *event = new_button_event();
+
+	event->key_id = (uint16_t)atoi(argv[1]);
+	event->pressed = *argv[2] != '0';
+	APP_EVENT_SUBMIT(event);
+
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(sub_caf_shell_command,
+	SHELL_CMD_ARG(set_button, NULL, "Simulate a button event with button ID and state",
+				set_button_handler, 3, 0),
+	SHELL_SUBCMD_SET_END
+);
+
+SHELL_CMD_REGISTER(caf, &sub_caf_shell_command, "CAF Shell commands", NULL);
```

Comment:
Maybe we can call the set `caf_events` (`Submit a CAF event with user-defined parameters` as help). Then we could call each command simply by an event name (e.g. `button_event`).